core/src/main/java/org/elasticsearch/index/translog/MultiSnapshot.java

```diff
@@ -99,60 +100,62 @@ public void close() throws IOException {
 
         boolean getAndSet(int index) {
```

Comment:
> This way this can inherit from BitSet and be used as a drop-in replacement for it.

I agree. I move this class to the common package and implement methods from BitSet.

> I don't think we need the getAndSet semantics any more. We can just have set and if people want to pre-get, they can.

The method `getAndSet` is optimized to avoid looking an entry twice, but I am fine to remove it.